src/components/input/input-password.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+import InputDefault from './input-default';
+import './input-password.scss';
+
+class InputPassword extends InputDefault {
+  constructor(props) {
+    super(props);
+
+    this.state = Object.assign({}, this.state, {
+      showPassword: !!props.showPassword,
+      hasAddon: true,
+    });
+
+    this.renderPasswordToggle = this.renderPasswordToggle.bind(this);
+    this.onPasswordToggleChange = this.onPasswordToggleChange.bind(this);
+  }
+
+  // TODO: How to handle this when there's already a componentWillReceiveProps method?
```

Comment:
You can do `super.componentWillReceiveProps()` if you need to call parents class method.